binary_search_tree/tree.py

```diff
@@ -1,3 +1,7 @@
+from logging import root
+import queue
```

Comment:
Unless I'm missing something it doesn't look like these `import`s get used, so you can remove them.